src/cart/crt_tree.c

```diff
@@ -93,27 +88,26 @@ crt_get_filtered_grp_rank_list(struct crt_grp_priv *grp_priv, uint32_t grp_ver,
 			D_GOTO(out, rc = 0);
 		}
 
-		*allocated = true;
 		*grp_size = grp_rank_list->rl_nr;
 	}
 
-
-	pri_root = grp_priv_get_primary_rank(grp_priv, root);
-	pri_self = grp_priv_get_primary_rank(grp_priv, self);
-
-	rc = d_idx_in_rank_list(grp_rank_list, pri_root, grp_root);
+	rc = d_idx_in_rank_list(grp_rank_list, root, grp_root);
 	if (rc != 0) {
 		D_ERROR("d_idx_in_rank_list (group %s, rank %d), "
 			"failed, rc: %d.\n", grp_priv->gp_pub.cg_grpid,
 			root, rc);
+		d_rank_list_free(grp_rank_list);
 		D_GOTO(out, rc);
 	}
 
-	rc = d_idx_in_rank_list(grp_rank_list, pri_self, grp_self);
-	if (rc != 0)
+	rc = d_idx_in_rank_list(grp_rank_list, self, grp_self);
+	if (rc != 0) {
 		D_ERROR("d_idx_in_rank_list (group %s, rank %d), "
 			"failed, rc: %d.\n", grp_priv->gp_pub.cg_grpid,
 			self, rc);
+		d_rank_list_free(grp_rank_list);
+		D_GOTO(out, rc);
```

Comment:
This line doesn't seem to do a lot.

Reply:
This is to avoid situation later when code is added after if () clause causing error case to execute stuff that it should not.